aya/src/programs/lirc_mode2.rs

```diff
@@ -59,6 +62,17 @@ impl LircMode2 {
     ///
     /// The returned value can be used to detach, see [LircMode2::detach].
     pub fn attach<T: AsFd>(&mut self, lircdev: T) -> Result<LircLinkId, ProgramError> {
+        self.attach_with_flag(lircdev, CgroupAttachFlag::None)
+    }
+
+    /// Attaches the program to the given lirc device.
+    ///
+    /// The returned value can be used to detach, see [LircMode2::detach].
+    pub fn attach_with_flag<T: AsFd>(
```

Comment:
Here and everywhere else: attach_with_cgroup_mode?

I'm not 100% convinced on mode, do you have a better name?

Reply:
Ha, now onto the hard part, naming :)

So far I think you've picked a good name.  Others I thought of are `CgroupAttach- Strategy or Policy or Cardinality or Rule`, they each have pros and cons.  ("Strategy" is a bit generic, "Policy" is good but feels more security focused, "Cardinality" is accurate but a bit long, "Rule" I kind of like.)

If we go with any of these, I'm thinking of changing the name of the default enum value, as a mode of `None` sounds odd.  Do these value names work?

```
pub enum CgroupAttachMode {
	Single,
	AllowMultiple,
	AllowOverride,
}
```